### PR TITLE
Adds a document for Profiling and an unmangler.

### PR DESCRIPTION
**Pull request by stevej**
Problem:
  Profiling needs to be documented so others can do it.

Solution:
  A document giving a step-by-step description of how to profile.

### CODE REVIEW

File: PROFILING.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+h1. Profiling linkerd-tcp
```
Comment (klingerf) [inline PROFILING.md]: Tiny nit: I forget which syntax uses `h1.`, but it's not markdown :)

screen shot 2017-04-13 at 4 51 02 pm

File: PROFILING.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+SVG viewer like [Gapplin](https://itunes.apple.com/us/app/gapplin/id768053424?mt=12)
+for the Mac.
+
+
+
+Footnotes
+---------
+
+#### [1] Setting a Useful Sample Rate
+When setting a sample rate (`-F`), remember that Nyquist theorem tells us that to see see things that happen with a rate of N you'll need to sample with a rate of 2N; It's useful to use `perf record` with a variety of different sampling rates while tracking down performance issues.
```
Comment (klingerf) [inline PROFILING.md]: Typo here: "see see". Also, mind wrapping this at 80 characters?

File: PROFILING.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+```
+
+Send it traffic through your favorite load tester.
+```
+slow_cooker_linux_amd64 -H 'default' -totalRequests 1000000 -qps 10 -concurrency 100 http://proxy-test-4e:7474/
+```
+
+Once your load test is complete, use the linkerd-tcp admin port to shutdown the process
+```
+curl -d "hello from buoyant" http://localhost:9989/shutdown
```
Comment (klingerf) [inline PROFILING.md]: I think `curl -X POST http://localhost:9989/shutdown` is a bit more canonical.

File: scripts/rust-unmangle.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
+#!/bin/sed -rf
+# Unmangle Rust symbols
+# Borrowed from https://github.com/Yamakaky/rust-unmangle
+# See https://git.kernel.org/cgit/linux/kernel/git/tip/tip.git/commit/?id=cae15db74999edb96dd9f5bbd4d55849391dd92b
+# Example, with [FlameGraph](https://github.com/brendangregg/FlameGraph):
+#     perf record -g target/debug/bin
+#     perf script | stackcollapse-perf | rust-unmangle | flamegraph > perf.svg
```
Comment (klingerf) [inline scripts/rust-unmangle.sh]: `rust-unmangle.sh`?

File: PROFILING.md (hunk 1 above)
Comment (stevej) [inline PROFILING.md]: oh lol, that's textile.

**Comment by stevej** [comment]
Ok, fixed those issues!

File: PROFILING.md (hunk 1 above)
Comment (klingerf) [inline PROFILING.md]: Hmm, I still see this `h1.` header in the latest commit. Maybe your change didn't get pushed?

**Comment by stevej** [comment]
Kevin, I'm sorry, I left the comment about fixing the `h1.` in the file but forgot to fix the actual file.

**Pull request merged**